libraries/AP_Motors/AP_MotorsHeli_RSC.cpp

```diff
@@ -418,3 +424,70 @@ float AP_MotorsHeli_RSC::calculate_desired_throttle(float collective_in)
     return throttle;
 
 }
+
+// autothrottle_run - calculate throttle output for governor controlled throttle
+void AP_MotorsHeli_RSC::autothrottle_run()
+{
+    float throttlecurve = calculate_throttlecurve(_collective_in);
+
+    // autothrottle main power loop with governor
+    if (_governor_engage && !_governor_fault) {
+        // calculate droop - difference between actual and desired speed
+        float governor_droop = (_governor_rpm - _rotor_rpm) * get_governor_droop_response();
+        _governor_output = governor_droop + ((throttlecurve - _governor_torque_reference) *  get_governor_ff());
+        if (_rotor_rpm < (_governor_rpm - 2.0f)) {
+            _governor_torque_reference += get_governor_compensator();   // torque compensator
+        } else if (_rotor_rpm > (_governor_rpm + 2.0f)) {
+            _governor_torque_reference -= get_governor_compensator();
+        }
+    // throttle output uses droop + torque compensation to maintain proper rotor speed
+    _control_output = constrain_float((_governor_torque_reference + _governor_output), throttlecurve * get_governor_ff(), 1.0f);
+        // governor and speed sensor fault detection - must maintain Rrpm -3/+2%
+        // speed fault detector will allow a fault to persist for 200 contiguous governor updates
+        // this is failsafe for bad speed sensor or severely mis-adjusted governor
+        if ((_rotor_rpm <= (_governor_rpm * 0.97f)) || (_rotor_rpm >= (_governor_rpm * 1.02f))) {
+            _governor_fault_count += 1.0f;
+            if (_governor_fault_count > 200.0f) {
+                _governor_fault = true;
+                _governor_engage = false;
+                _governor_output = 0.0f;
```

Comment:
Matt, there is only two variables that could be reset in such a function. The flags can't be in it, because they depend on the status on the internal fault checking and spool logic. By the time you declare the function, write the function, and put in calls to it in four places it is more lines of code than just inline-resetting the two variables in four places. That's why I didn't do it.

Reply:
done.  Thanks!